bundles/core/src/test/java/com/adobe/cq/wcm/core/components/internal/models/v2/PageImplTest.java

```diff
@@ -87,6 +97,39 @@ public void testPage() throws ParseException {
         Utils.testJSONExport(page, Utils.getTestExporterJSONPath(TEST_BASE, PAGE));
     }
 
+    @Test
+    public void testChildren() {
+        Page page = getPageUnderTest(PAGE);
+        ModelFactory modelFactory = Mockito.mock(ModelFactory.class);
+        Whitebox.setInternalState(page, "modelFactory", modelFactory);
+
+        Page childPage = getPageUnderTest(PAGE_CHILD);
+        when(modelFactory.getModelFromWrappedRequest(any(SlingHttpServletRequest.class), any(Resource.class), eq(Page.class))).thenReturn(childPage);
+
+        Map<String, ? extends HierarchyNodeExporter> children = page.getExportedChildren();
+        Assert.assertEquals(2, children.size());
+        Assert.assertEquals(childPage, children.get(PAGE_CHILD));
+
+    }
+    @Test
+    public void testGetRootUrl() {
+
+        // no parent
+        Page rootPage = getPageUnderTest(PAGE);
+        Assert.assertEquals(CONTEXT_PATH + PAGE + ".model.json", rootPage.getRootUrl());
+
+        // with parent
+        Page childPage = getPageUnderTest(PAGE_CHILD);
+        Assert.assertEquals(CONTEXT_PATH + PAGE + ".model.json", childPage.getRootUrl());
+    }
+
+    @Test
+    public void testGetRootModel() {
+        // no parent
+        Page page = getPageUnderTest(PAGE);
+        Assert.assertEquals(page, page.getRootModel());
```

Comment:
Eventually, a test of the root model starting from the child that equals the parent page?